Automattic/pocket-casts-android
# Bookmarking syncing

## Description

This change adds bookmarking syncing. As this isn't available on the server yet we won't be able to fully test if it works yet. 

@srjakes would you mind checking over this?

## Testing Instructions

1. Edit the BookmarksViewModel class to add a bookmark when opening the page.
```
bookmarkManager.add(episode, 42, "My bookmark")
```
2. Deploy the app
3. Sign in with a Plus account
4. Play an episode
5. In Android Studio open the "App Inspection" window and connect to the process
6. Open the full screen player and tap the bookmarks tap to add the bookmark
7. Go to the Profile tab
8. Tap "Refresh now"
9. Look in the "Network Inspector" and looks for the `/sync/update` call. 
10. ✅ Verify it has sent a 'UserBookmark' in the request to the server by searching logcat for `UserBookmark`.
